Update to python 3

ed588 opened the pull request:
I've updated the code from python 2 to python 3. Initially I used the 2to3 tool, but I've had do make quite a few manual changes involving the separation of `str` and `bytes` that is new in python 3. I'm not quite sure that everything here works yet, but the main functionality seems to be working fine.

ed588 committed:
windows doesn't like colons in filenames, and i'm stuck on windows sadly

ed588 committed:
update gitignore

ed588 committed:
run 2to3 on main file

ed588 committed:
add a requirements.txt

ed588 committed:
simplify where the data goes, at the expense of a small amount of functionality

ed588 committed:
remove gz files

ed588 committed:
port to py3

ed588 committed:
strings and bytes are no longer the same

ed588 committed:
remove print statements

ed588 committed:
remove comments and unneeded import

ed588 commented:
Just to clarify - I haven't yet tested all the functionality, but what I *have* tested so far works without any annoying `str`/`bytes` bugs.

Eluvatar commented:
This is great! I'll take a closer look after z-day so we can get it merged
:)

On Tue, Oct 30, 2018, 11:40 Calvin Edwards <notifications@github.com> wrote:

> Just to clarify - I haven't yet tested all the functionality, but what I
> *have* tested so far works without any annoying str/bytes bugs.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Eluvatar/mock-nationstates/pull/2#issuecomment-434350319>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABBxIhBVk9squzJC9tGaoLko1-PWvNU6ks5uqHLUgaJpZM4YByKC>
> .
>

ed588 committed:
Remove cherrypy dependencies from requirements.txt

The requirements in the file appeared to be for the windows-specific
version of cherrypy, which obviously won't work on other systems.
Installing from the requirements file will install dependencies anyway
so this shouldn't actually change anything.

ed588 committed:
Remove gitignore, and add the xml files

I'm not 100% sure why this was set up like this initially,
but doing it like this is simpler, without making it any more difficult
to put your own data in if you want.

Plus, I can put it back to how it was later, but at the moment it is
easier for me to test this with the xml files like this.

Eluvatar commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
 *~
 *.pyc
+venv/
+.vscode/
```
```suggestion
.vscode/
```

Eluvatar commented on data/.gitignore:
```diff
@@ -1,3 +0,0 @@
-???ions-????????.xml
```
Please undelete this, I found it useful to put assorted XML files in the data directory and not be bothered by git.

Further, `telegrams.json` may contain your actual TG ID and keys, so you don't want to check that in by accident!

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
+        namej = mm.find(b"</NAME>",namei)
         name = mm[namei:namej]
-        idx[id_str(name)] = (i,j)
+        idx[id_str(name.decode())] = (i,j)
```
```suggestion
        idx[id_str(name)] = (i,j)
```

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
-        namej = mm.find("</NAME>",namei)
+        namei = mm.find(b"<NAME>",i)+len(b"<NAME>")
+        namej = mm.find(b"</NAME>",namei)
         name = mm[namei:namej]
```
```suggestion
         name = mm[namei:namej].decode()
```

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
 scan(rm, "<REGION>", "</REGIONS>", regions)
 
 def event_scan(mm,idx):
     beg = "<EVENT"
```
```suggestion
     beg = "<EVENT".encode()
```

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
 
 def event_scan(mm,idx):
     beg = "<EVENT"
     end = "</EVENT>"
```
```suggestion
     end = "</EVENT>".encode()
```

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
 def event_scan(mm,idx):
     beg = "<EVENT"
     end = "</EVENT>"
+    beg = beg.encode()
```
```suggestion
```

Eluvatar commented on mock_server.py:
```diff
@@ -74,36 +74,39 @@ def extract(mm,idx,k):
     beg = "<EVENT"
     end = "</EVENT>"
+    beg = beg.encode()
+    end = end.encode()
```
```suggestion
```

Eluvatar commented on mock_server.py:
```diff
@@ -214,7 +217,7 @@ def world_api_result(nm,nations,rm,regions,em,events,q,params):
         hroot = ET.Element("HAPPENINGS")
         root.append(hroot)
-        for eid in xrange(beforeid,sinceid,-1):
+        for eid in range(beforeid,sinceid,-1):
```
I wish there was a nice way to be backwards compatible and fast. Oh well.

Eluvatar commented on mock_server.py:
```diff
@@ -384,4 +387,3 @@ def default(self, *args, **params):
     cherrypy.tree.mount(MockNationStatesApi(),'/cgi-bin/api.cgi',conf)
     cherrypy.tree.mount(None,'/',staticconf)
     cherrypy.engine.start()
-    cherrypy.engine.block()
```
Why cut this line?

Eluvatar commented on ns.py:
```diff
@@ -13,10 +13,5 @@
-  i,i2=codecs.getencoder('ascii')(string.lower(name))
-  id=i.translate(__id_str__trans)
-  return id
+  return str(name.lower().replace(" ", "_"))
```
:+1:

Eluvatar commented on data/happenings.xml:
```diff
@@ -1 +1,13710 @@
-happenings-20131125:26.xml
\ No newline at end of file
+<WORLD><HAPPENINGS><EVENT id="7975081">
```
You appear to have changed the symlink to point to the contents of the file, instead of to the file. This doesn't work for me when I check it out.

Do symlinks checked out by git just not work right on windows?

Eluvatar reviewed (changes requested):
I've asked for a few minor changes, including one that simply breaks it for me (at least on linux).

Please address them, and update the README, and I will happily merge your change.
